caps/capabilities.go

```diff
@@ -33,12 +33,63 @@ type Capability struct {
 	// the behavior allowed and expected from providers and consumers of that
 	// capability, and also which information should be exchanged by these
 	// parties.
-	Type Type
+	Type *Type
 	// Attrs are key-value pairs that provide type-specific capability details.
 	Attrs map[string]string
 }
 
+var (
+	testCapability = &Capability{
+		Name:  "test-name",
+		Label: "test-label",
+		Type:  testType,
+		Attrs: nil,
+	}
+)
+
 // String representation of a capability.
 func (c Capability) String() string {
 	return c.Name
 }
+
+// CapabilityRepr is the JSON representation of a capability.
+// It exists so that Type can be replaced by Type.Name as we don't want to
+// create or describe capabilities fully, just to refer to them.
+type CapabilityRepr struct {
+	// Name is a key that identifies the capability. It must be unique within
+	// its context, which may be either a snap or a snappy runtime.
+	Name string `json:"name"`
+	// Label provides an optional title for the capability to help a human tell
+	// which physical device this capability is referring to. It might say
+	// "Front USB", or "Green Serial Port", for example.
+	Label string `json:"label"`
+	// Type defines the type of this capability. The capability type defines
+	// the behavior allowed and expected from providers and consumers of that
+	// capability, and also which information should be exchanged by these
+	// parties.
+	TypeName string `json:"type_name"`
+	// Attrs are key-value pairs that provide type-specific capability details.
+	Attrs map[string]string `json:"attrs"`
+}
+
+// ConvertToRepr makes a CapabilityRepr from a Capability.
+// This function is useful for creating JSON representation of a capability
```

Comment:
Capability itself should be representable as JSON, if we care about it. Please see:

    https://golang.org/pkg/encoding/json/#Marshaler